fabric-events-player-v1/src/main/java/net/fabricmc/api/event/player/v1/PlayerRespawnCallback.java

```diff
@@ -0,0 +1,34 @@
+package net.fabricmc.api.event.player.v1;
+
+import net.minecraft.server.network.ServerPlayerEntity;
+import net.minecraft.world.dimension.DimensionType;
+
+import net.fabricmc.fabric.api.event.Event;
+import net.fabricmc.fabric.api.event.EventFactory;
+
+/**
+ * Represents a callback for when a {@link ServerPlayerEntity} is respawned.
+ *
+ * <p>This occurs when a player dies, or changes dimensions, such as returning to the overworld from the end.
+ *
+ * <p>Note the player can be repositioned.
+ */
+public interface PlayerRespawnCallback {
+	Event<PlayerRespawnCallback> EVENT = EventFactory.createArrayBacked(PlayerRespawnCallback.class, (callbacks) -> (newPlayer, oldPlayer, newDimension, alive) -> {
+		for (PlayerRespawnCallback callback : callbacks) {
+			callback.onRespawn(newPlayer, oldPlayer, newDimension, alive);
+		}
+	});
+
+	/**
+	 * Called when a player respawns.
+	 *
+	 * <p>Note the player can be repositioned.
```

Comment:
```suggestion
	 * <p>When this method is called, {@code newPlayer} will be at its spawn position. 
	 * It may be repositioned by callbacks to change where it respawns. 
	 * {@code oldPlayer}'s coordinates and world will  be the same as before the respawn, 
	 * however its {@linkplain net.minecraft.entity.Entity#dimension dimension} will have been set to {@link net.minecraft.world.dimension.DimensionType#OVERWORLD}.
	 *
	 * <p> Player data will also have been copied from {@code oldPlayer} to {@code newPlayer}.
	 * The actual copied data depends on various factors such as the value of {@code isAlive} or of the {@code keepInventory} gamerule.
```
(should probably reformat and import packages)